src/tsickle.ts

```diff
@@ -592,18 +605,56 @@ class Annotator extends ClosureRewriter {
   }
 
   /**
+   * workaround for https://github.com/Microsoft/TypeScript/issues/12597
+   */
+  private writeModuleSpecifier(moduleSpecifier: ts.Expression) {
+    function getText() {
+      switch (moduleSpecifier.kind) {
+        case ts.SyntaxKind.StringLiteral:
+          return (moduleSpecifier as ts.StringLiteral).text;
+        default:
+          // The above list of kinds should be exhaustive.
+          throw new Error(`unhandled moduleSpecifier kind: ${ts.SyntaxKind[moduleSpecifier.kind]}`);
```

Comment:
This could be simplified to something like:

```
if (moduleSpecifier.kind !== ts.SyntaxKind.StringLiteral) throw new Error ...
let moduleId = (moduleSpecifier as ts.SyntaxKind.StringLiteral).text